prov/util/src/rxm_av.c

```diff
@@ -55,10 +54,16 @@ void *rxm_av_alloc_conn(struct rxm_av *av)
 	return conn_ctx;
 }
 
+void _rxm_av_free_conn(void *conn_ctx)
+{
+	/* assert(ofi_mutex_held(&av->util_av.lock)); */
```

Comment:
If a lock must be held by a function, keep the assert active.  It documents the requirement and forces the issue.